### ISSUE TITLE
Options for running commands on a specified range of versions of cargo

### ISSUE THREAD

**Issue by taiki-e** [taiki-e/cargo-hack#93]
This may be useful for catching issues like https://github.com/BurntSushi/termcolor/issues/35, https://github.com/rust-lang/regex/issues/685, https://github.com/rayon-rs/rayon/pull/761#discussion_r428639158, https://github.com/rust-lang/rust-clippy/issues/6324. 

## Examples

something like:

```
$ cargo hack check --version-range 1.36..stable
info: running `cargo +1.36.0 check`
...
info: running `cargo +1.37.0 check`
...
...
info: running `cargo +1.47.0 check`
...
info: running `cargo +1.48.0 check` (current stable)
...
```

with specified step:

```
$ cargo hack check --version-range 1.36..stable --version-step 2
info: running `cargo +1.36.0 check`
...
info: running `cargo +1.38.0 check`
...
...
info: running `cargo +1.46.0 check`
...
info: running `cargo +1.48.0 check` (current stable)
...
```

**Comment by jhpratt** [comment]
Would this take point releases into account? It's probably not strictly necessary.

**Comment by taiki-e** [comment]
I would like to use rustup's [`major.minor` channels](https://blog.rust-lang.org/2020/11/27/Rustup-1.23.0.html#support-for-installing-minor-releases). (i.e., point releases do not take into account, and always select the latest patch version of each minor version)

**Comment by jhpratt** [comment]
Ah, advantages of the new rustup release!

**Comment by taiki-e** [comment]
Published in 0.5.0 (#102)
